ita-social-projects/GreenCityUBS
# #5313 Fix duplicate bags from '/order-details' endpoint

## JIRA

* [Main JIRA ticket](https://github.com/ita-social-projects/GreenCity/issues/5313)


## Code reviewers

- [x] @juseti 
- [x] @LiliaMokhnatska 
- [ ] @NKorzhik 
- [ ] @ABbondar 
- [ ] @sergoliarnik 
- [ ] @BranetE 
- [ ] @Markiro1 
- [x] @OlegVat 

## Summary of issue

Services are duplicated in section 'Services' on the Step 1 of Order form

## Summary of change

1) added new changelog for creating foreign key locationId to OrderAddress
2) added OneToMany relationship between Location and OrderAddress
3) separated '/order-details' endpoint into 2:
'/order-details-for-tariff'
'/details-for-existing-order/{orderId}'
4) separated method getFirstPageData into 2:
- getFirstPageDataByTariffAndLocationId()
- getFirstPageDataByOrderId()
5) 
- added tests for getFirstPageDataByOrderId() and getFirstPageDataByTariffAndLocationId() methods in UBSClientServiceImplTest
- changed some methods for new logic
6) added static import variables of ErrorMessage class
7) excluded orderAddresses field from @EqualsAndHashCode() and @ToString() annotations in Location
8) excluded location field from @EqualsAndHashCode() and @ToString() annotations in OrderAddress
9) added documentation

## Testing approach

1. Sign in as user
2. Go to https://www.testgreencity.ga/GreenCityClient/#/ubs
3. Open the page 'UBS-courier'
4. Click on button 'Order'
5. Each service should be displayed only once in section 'Services' on the Step 1 of Order form

## CHECK LIST
